File: 디자인패턴/java 디자인 패턴/Observer/EvenBettingPlayer.java
public class EvenBettingPlayer extends Player {
    protected EvenBettingPlayer(String name) {
        super(name);
    }

    @Override
    protected void ProcessDiceBetting(int Dice_Num) {
        System.out.println("\nPlayer: " + getPlayer_name());
        if (Dice_Num%2==0) System.out.println("Win!");
        else System.out.println("Lose!");
    }
}

File: 디자인패턴/Python 디자인 패턴/Observer/EvenBettingPlayer.py
from Player import*
class EvenBettingPlayer(Player):
    def __init__(this, name: str) -> None:
        super().__init__(name)
    
    def Update(this, diceNum: int) -> None:
        print(f"\nPlayer: {this.getPlayerName()}")
        if(diceNum%2==0):print("Win")
        else:print("Lose!")
        pass
    pass